import "VisionObservations.proto";

message WorldInfo{
	required RobotPose myPosition = 1 ;
	repeated Ball Balls = 2;
	repeated RobotPose OtherRobots = 3;
	repeated UnknownObjects UFOs = 4 ;
	repeated Ball GlobBalls = 5;
}

message SharedWorldInfo{
	required RobotPose playerClosestToBall = 1 ;
	repeated TeammatePose teammatePosition = 2;
}

message TeammatePose{
	required RobotPose pose = 1;
	required uint32 robotId = 2;
}

message RobotPose{
	required float X = 1 [default = -100000];
	required float Y = 2 [default = -100000];
	required float phi = 3 [default = -100000];
	required float confidence = 4 [default = -100000];
}

message Ball{
	required float relativeX = 1 [default = -100000];
	required float relativeY = 2 [default = -100000];
	required float relativeXspeed = 3 [default = -100000];
	required float relativeYspeed = 4 [default = -100000];

	required float varianceX = 5 [default = -100000];
	required float varianceY = 6 [default = -100000];
	required float varianceXspeed = 7 [default = -100000];
	required float varianceYspeed = 8 [default = -100000];
}

message UnknownObjects{
	optional float relativeX = 1 [default = -100000];
	optional float relativeY = 2 [default = -100000];
	optional float height = 3 [default = -100000];
	optional float relativeXspeed = 4 [default = -100000];
	optional float relativeYspeed = 5 [default = -100000];
}

message LocalizationResetMessage{
	enum RESET{
		UNIFORM=0;
		READY=1;
		SET=2;
		PENALISED=3;
		MANUAL=4; 
	}
	required int32 type = 1; // RoboCup GameController Data PLAYER_SET, PLAYER_READY, PLAYER_PENALISED
	required bool kickOff = 2;
	required float xPos = 3 [default = 0];
	required float yPos = 4 [default = 0];
	required float phiPos = 5 [default = 0];
}

message LocalizationData{
	required WorldInfo World= 1;
	repeated RobotPose Particles = 2;
	required RobotPose RobotPosition = 3; //NaoqiRobotPosition
	required ObservationMessage Observations = 4;
}

message header{
	required sint32 NextMsgByteSize =1 [default = -1];
	required bytes NextMsgName = 3 [default = "Undef" ];
	
}

message LocalizationDataForGUI{
	repeated RobotPose Particles = 1;
}
